Add a workflow to trigger E2E tests with the "/run-e2e" command

enescakir opened the pull request:
Currently, to trigger E2E tests, we need to open the workflow page on the GitHub UI, select a branch, and run the tests. There's a more developer-friendly way to do this. We can easily add support for the `/run-e2e` command using GitHub Actions workflows. When you run `/run-e2e` on a pull request, it triggers E2E tests for that PR. You can also pass specific test cases with `/run-e2e
vm,github_runner_ubuntu_2404`.

It only accepts comments from contributors, similar to the E2E test trigger.

Since "issue_comment" event doesn't have pull request head ref, we get it from API.

Additionally, we can add a check to this workflow to trigger E2E tests if there are any code changes in the `rhizome` path.
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds a GitHub Actions workflow to trigger E2E tests via `/run-e2e` comment on PRs, with contributor validation and test case security checks.
> 
>   - **New Workflow**:
>     - Adds `trigger-e2e.yml` to trigger E2E tests via `/run-e2e` comment on PRs.
>     - Validates comment author is a contributor and test cases string is secure.
>     - Uses `actions/github-script@v7` to handle comment parsing and validation.
>   - **Behavior**:
>     - Triggers E2E tests for PRs when `/run-e2e` is commented.
>     - Supports specific test cases via `/run-e2e vm,github_runner_ubuntu_2404`.
>     - Adds a reaction to the comment based on validation success.
>   - **Modifications**:
>     - Updates `e2e.yml` to handle `workflow_dispatch` with specific test cases input.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=ubicloud%2Fubicloud&utm_source=github&utm_medium=referral)<sup> for bb327adea054686b6b016c763670ed6593e2d0d8. You can [customize](https://app.ellipsis.dev/ubicloud/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

enescakir commented:
/run-e2e

enescakir commented on .github/workflows/trigger-e2e.yml:
```diff
@@ -0,0 +1,38 @@
+              })
+
+              // Create a workflow dispatch event
+              const testCases = context.payload.comment.body.split(' ')[1]
```
We don't have tests cases with spaces

jeremyevans reviewed (commented):
I'm not sure if there are any security considerations here with the actions and pull requests write permissions. If `inputs.test_cases` contains malicious code, what are the implications?  Can anyone who comments trigger the workflow?  Is there a way to limit this to only github organization members?

enescakir commented:
> I'm not sure if there are any security considerations here with the actions and pull requests write permissions. If `inputs.test_cases` contains malicious code, what are the implications? Can anyone who comments trigger the workflow? Is there a way to limit this to only github organization members?

Valid concerns.

I added a contributor check, so it only accepts comments from contributors, similar to the E2E test trigger.

I also added the check `const isValid = /^[a-zA-Z0-9-,]+$/.test(testCases);` to ensure that test cases can only be alphanumeric, commas, or hyphens.

enescakir commented on .github/workflows/trigger-e2e.yml:
```diff
@@ -0,0 +1,58 @@
+              // Check if the given test cases string is secure or empty
+              const testCases = context.payload.comment.body.split(' ')[1]
+              console.log(`testCases: ${testCases}`);
+              const isValid = /^[a-zA-Z0-9-,]+$/.test(testCases);
```
Good catch

jeremyevans reviewed (approved):
This looks OK to me now, after the issues mentioned by ellipsis are addressed.  Might be worth having someone else review as well, as this isn't my area of expertise.
